### PR TITLE
feat/expert-login

### PR DESCRIPTION
**Pull request by hemal-digiflux**

### CODE REVIEW

File: src/redux/reducers/server-reducer/server.reducer.ts [hunk 1]
```diff
@@ -39,13 +39,16 @@ export const ServerReducer: Reducer<ServerStateInterface> = (
     state: ServerStateInterface = ServerInitialState,
     action: ServerActionTypes,
 ) => {
+    console.log(action);
```
Comment (Diego-Romero) [inline src/redux/reducers/server-reducer/server.reducer.ts]: remove pls

File: src/interfaces/responses/expert-login-response.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface ExpertLoginResponse {
```
Comment (Diego-Romero) [inline src/interfaces/responses/expert-login-response.ts]: It feels like we have many many interfaces doing the same thing, could we please refactor all of these onto a single interface that is the generic response from the server?

File: src/redux/actions/server/server.actions.test.ts [hunk 3]
```diff
@@ -140,4 +142,39 @@ describe('server actions', () => {
...
+                expertise: ['PERSONAL_COACH', 'YOGA_COACH'],
+                phone: '123123123',
+                pictureUrl:
+                    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSBx76lQwzW2cAsjz5JqgVp_ReTpVji6G_pMO6crXSJn9NETq3F',
+            };
+            const expectedAction = {
+                type: ServerActions.SET_EXPERT_USER,
+                expertUser: expertUserMeResponse,
+            };
+            expect(setExpertUserAction(expertUserMeResponse)).toEqual(expectedAction);
```
Comment (Diego-Romero) [inline src/redux/actions/server/server.actions.test.ts]: Do you have intelisense in your Editor? this is throwing me an error in my code editor, the types don't match

File: src/redux/actions/server/server.actions.test.ts [hunk 4]
```diff
@@ -140,4 +142,39 @@ describe('server actions', () => {
...
+        });
+    });
+
+    describe('setExpertUserAction', () => {
+        it('should return the correct type and data', () => {
+            const expertUserMeResponse = {
+                _id: '123123',
+                email: 'expert@fitmind.io',
+                name: 'Expert1',
+                createdAt: new Date().toISOString().slice(0, 10),
```
Comment (Diego-Romero) [inline src/redux/actions/server/server.actions.test.ts]: why do we need this slice(0,10)?

File: src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    lastName: 'User',
+    createdAt: new Date().toISOString().slice(0, 10),
+    approvedStatus: 'APPROVED',
+    description:
+        'II’m a versatile personal trainer, who has experience in Functional training, wieight training and yoga. I’ve been training people of all ages for over 12 years improving their overall quality of life.',
+    email: 'expert1@fitmind.io',
+    expertise: ['PERSONAL_COACH', 'YOGA_COACH'],
+    phone: '123123132',
+    weeklyAvailability: {
+        monday: ['00:00-00:30'],
```
Comment (Diego-Romero) [inline src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts]: This formatting is not correct, please check the server for a reference

File: src/redux/actions/server/server.actions.test.ts (hunk 4 above)
Comment (hemal-digiflux) [reply src/redux/actions/server/server.actions.test.ts]: This gives output like "2019-08-27", and it means only the date not time.. So initially I might have referred some doc/swagger which made me do this. Do you think we need to keep the full date+ time?

File: src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts (hunk 5 above)
Comment (hemal-digiflux) [reply src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts]: As last time on call, we thought to have 30 mins timeslots, So I kept that complete 30 mins slots in response.. Do you think we need to identify the slot only by start time?  Then I will revert this to the older one/swagger contains

File: src/interfaces/responses/expert-login-response.ts (hunk 2 above)
Comment (hemal-digiflux) [reply src/interfaces/responses/expert-login-response.ts]: That could be one option to create one common response type , which will have this "message" : string. But in the other hand, when we will make changes in any of this responses at that time we need to create separate ones. So better to keep them separate, even though we have some duplication. Thats just my view. Please let me know.

File: src/redux/reducers/server-reducer/server.reducer.ts (hunk 1 above)
Comment (hemal-digiflux) [reply src/redux/reducers/server-reducer/server.reducer.ts]: Done!

File: src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts (hunk 5 above)
Comment (hemal-digiflux) [reply src/redux/reducers/server-reducer/server-example-responses/expert-me-example-response.ts]: Changed as per swagger.

File: src/interfaces/responses/expert-login-response.ts (hunk 2 above)
Comment (Diego-Romero) [reply src/interfaces/responses/expert-login-response.ts]: I think having separate ones makes sense when the responses are different, but having 10 interfaces with the exact same field doesn't makes much sense to me, I'm not too fuzzed though.

File: src/redux/actions/server/server.actions.test.ts (hunk 4 above)
Comment (Diego-Romero) [reply src/redux/actions/server/server.actions.test.ts]: Ah I see, ok no worries in this case.

File: src/redux/reducers/server-reducer/server.reducer.ts (hunk 1 above)
Comment (Diego-Romero) [reply src/redux/reducers/server-reducer/server.reducer.ts]: This is not done, I can still see it...

File: src/redux/actions/server/server.actions.test.ts (hunk 3 above)
Comment (Diego-Romero) [reply src/redux/actions/server/server.actions.test.ts]: ?

File: src/redux/reducers/server-reducer/server.reducer.ts (hunk 1 above)
Comment (hemal-digiflux) [reply src/redux/reducers/server-reducer/server.reducer.ts]: Sorry, forgot to stage.

File: src/redux/actions/server/server.actions.test.ts (hunk 3 above)
Comment (hemal-digiflux) [reply src/redux/actions/server/server.actions.test.ts]: Yes, Its already on, and I have made change in this. Now you wont see the error

**Pull request merged**